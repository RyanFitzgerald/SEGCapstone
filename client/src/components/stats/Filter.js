import React from 'react';
import PikadayWrapper from '../PikadayWrapper';


const Filter = (props) => {
  let startDate, endDate, postalCode;

  function handleFilter() {
    // Send fields
    props.handleFilter(startDate, endDate, postalCode);
  }

  function resetFilter() {
    startDate.field.value = '';
    endDate.field.value = '';
    postalCode.value = '';

    // Call specific prop function
    props.resetFilter();
  }

  return (
    <div className="row">
      <div className="column">
        <h2 className="card-title">Filter Data</h2>
        <div className="card">
          <div className="row">
            <div className="sm-4 column">
              <label className="form-label" htmlFor="month">Start Date</label>
              <PikadayWrapper name="start-date" ref={input => startDate = input} id="start-date" onSelect={handleFilter} />
            </div>
            <div className="sm-4 column">
              <label className="form-label" htmlFor="year">End Date</label>
              <PikadayWrapper name="end-date" ref={input => endDate = input} id="end-date" onSelect={handleFilter} />
            </div>
            <div className="sm-4 column">
              <label className="form-label" htmlFor="postalCode">Postal Code</label>
              <input ref={input => postalCode = input} id="postalCode" name="postalCode" className="form-text" type="text" onKeyUp={handleFilter}/>
            </div>
          </div>
          <button className="advanced__toggle" id="advanced-toggle" onClick={resetFilter}>Reset Filter</button>
        </div>
      </div>
    </div>
  );
};

export default Filter;