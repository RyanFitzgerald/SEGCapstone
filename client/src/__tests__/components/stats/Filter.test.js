import React from 'react';
import Filter from '../../../Components/stats/Filter';

describe('<Filter />', () => {

  beforeAll(() => {
    // Add necessary info to user
    const user = {
      role: {
        level: 1
      }
    };
    localStorage.setItem('user', JSON.stringify(user));

  });

  it('should render correctly', () => {
    // Return empty array on requests
    mock.onAny().reply(200, []);
    // Create mock functions / spies
    const resetFilterMock = jest.fn();
    const handleFilterMock = jest.fn();

    // Shallow render it
    const wrapper = shallow(
      <Filter resetFilter={resetFilterMock} handleFilter={handleFilterMock}/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
