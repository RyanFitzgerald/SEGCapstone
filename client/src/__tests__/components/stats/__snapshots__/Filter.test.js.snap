// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render correctly 1`] = `
<div
  className="row"
>
  <div
    className="column"
  >
    <h2
      className="card-title"
    >
      Filter Data
    </h2>
    <div
      className="card"
    >
      <div
        className="row"
      >
        <div
          className="sm-4 column"
        >
          <label
            className="form-label"
            htmlFor="month"
          >
            Start Date
          </label>
          <PikadayWrapper
            id="start-date"
            name="start-date"
            onSelect={[Function]}
          />
        </div>
        <div
          className="sm-4 column"
        >
          <label
            className="form-label"
            htmlFor="year"
          >
            End Date
          </label>
          <PikadayWrapper
            id="end-date"
            name="end-date"
            onSelect={[Function]}
          />
        </div>
        <div
          className="sm-4 column"
        >
          <label
            className="form-label"
            htmlFor="postalCode"
          >
            Postal Code
          </label>
          <input
            className="form-text"
            id="postalCode"
            name="postalCode"
            onKeyUp={[Function]}
            type="text"
          />
        </div>
      </div>
      <button
        className="advanced__toggle"
        id="advanced-toggle"
        onClick={[Function]}
      >
        Reset Filter
      </button>
    </div>
  </div>
</div>
`;
